### DISCUSSION TITLE
Best way to build microservices and share auth between them.

### DISCUSSION THREAD

**Discussion by hillkim** [supabase/supabase#1847] [Questions] [answered]
Hello Awesome team! I am fairly new to supabase. I think it is great for any project. My question is , is there a way I can build microservices with independent postgres instances while sharing other resources like Storage and Auth between each one of them? Any guidance , thoughts and suggestions will be appreciated.

**Comment by awalias** [comment, marked as answer]
hey @hillkim 

the main thing is you want to make sure the same jwt_secret is known throughout all the microservices

from the dashboard:

in Settings > API

![image](https://user-images.githubusercontent.com/458736/121109303-dd856800-c83d-11eb-8cf0-c158575519e9.png)

this way every service can validate jwts generated by Supabase Auth like:

```js
var jwt = require('jsonwebtoken');
var decoded = jwt.verify(access_token, jwtSecret);

console.log(decoded)
```

is this what you had in mind?

**Reply by hillkim** [reply]
That makes sense @awalias .I will see it through. Thanks for the feedback.
